native/src/widget/scrollable.rs

```diff
@@ -389,10 +460,13 @@ where
 /// The local state of a [`Scrollable`].
 ///
 /// [`Scrollable`]: struct.Scrollable.html
-#[derive(Debug, Clone, Copy, Default)]
+#[derive(Debug, Default, Clone)]
 pub struct State {
     scroller_grabbed_at: Option<f32>,
-    offset: f32,
+    prev_offset: u32,
+    snap_to_bottom: bool,
+    offset: RefCell<f32>,
```

Comment:
Oooof, I think interior mutability is frowned upon in the Elm/iced ethos. Would it be plausible to refactor this into messages instead?

(Again @hecrj is the expert here)

Reply:
@hecrj mentioned using interior mutability in https://github.com/hecrj/iced/issues/307#issuecomment-626245954 and I myself didn't see a better way, so that's why I chose to do that.

Reply:
Ahhh yepp, nvm! ^^'